apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/TraceContext.java

```diff
@@ -712,7 +713,7 @@ public int hashCode() {
     }
 
     void setApplicationClassLoader(@Nullable ClassLoader classLoader) {
-        if (classLoader != null) {
+        if (classLoader != null && !ClassLoaderUtils.isAgentClassLoader(classLoader)) {
```

Comment:
`classLoader != null` test unnecessary now

Reply:
It _is_ necessary - if `classLoader == null`, then `!ClassLoaderUtils.isAgentClassLoader(classLoader)` would evaluate to `true` and enter the `if` block, which is not what we want.

Reply:
[minor] I think the expression would be easier to grasp if inverted and returning

```java
if (classLoader == null || ClassLoaderUtils.isAgentClassLoader(classLoader)) {
    return;
}
``` 

Or even


```java
if (ClassLoaderUtils.isBootstrapClassLoader(classLoader) || ClassLoaderUtils.isAgentClassLoader(classLoader)) {
    return;
}
```

Reply:
OK